x/capability/types/keys.go

```diff
@@ -39,7 +39,12 @@ func RevCapabilityKey(module, name string) []byte {
 // FwdCapabilityKey returns a forward lookup key for a given module and capability
 // reference.
 func FwdCapabilityKey(module string, cap *Capability) []byte {
-	return []byte(fmt.Sprintf("%s/fwd/%p", module, cap))
+	// truncate the key to fixed length, keep backward compatible on common architectures.
+	key := fmt.Sprintf("%#010p", cap)
+	if len(key) > 10 {
+		key = key[len(key)-10:]
+	}
+	return []byte(fmt.Sprintf("%s/fwd/0x%s", module, key))
```

Comment:
Why is this different from what we, recently, updated on `main` which is

```go
	return []byte(fmt.Sprintf("%s/fwd/%#016p", module, cap))
```

Reply:
This one is not breaking for common architectures, by using the same length as before, but for mac m1, it truncate the high byte, to avoid hash mismatch.

Reply:
But these keys are not in merkle state, so no change would be breaking AFAIK. As long as gas consumption is the same.

Reply:
exactly, the gas would be different if the encoded pointer has different length, the issue is here: https://github.com/cosmos/cosmos-sdk/issues/11726, it seems you have participated in it too.

Reply:
[ethermint feemarket](https://github.com/evmos/ethermint/blob/main/x/feemarket/keeper/abci.go#L56) will save the block gas used to merkle state ;D

Reply:
Yes, the solution to that issue is `return []byte(fmt.Sprintf("%s/fwd/%#016p", module, cap))`. Are you saying this wont work for you?

Reply:
it works, but it's a breaking one, so can't backport to old versions directly.

Reply:
Its breaking because it charges a different gas?

Reply:
yes

Reply:
This seems ok given the circumstances, but it'd be nice to indicate in the code this is a temporary patch and should not be replicated. Also, I think the tests should use the two pointer values in the referenced issue

Nice work and thanks for opening up a non-breaking fix

Reply:
added comments and backward compatibility test.